frontend/src/pages/components/WebSocketCall.jsx

```diff
@@ -20,25 +20,34 @@ export default function WebSocketCall({ socket }) {
     if (!message) {
       return;
     }
-    socket.emit("data", {'username': sessionStorage.Username + ": ", 'message': message});
+    socket.emit("message", {'token': sessionStorage.token, 'content': message, 'public_id': sessionStorage.currentChat});
     setMessage("");
   };
 
+  socket.emit("join", {'token': sessionStorage.token})
   useEffect(() => {
-    socket.on("data", (data) => {
-      setMessages([...messages, data.data]);
+      socket.on("join", (data) => {
+          console.log("received join response from socket");
+          console.log(data);
+      });
+  });
+  useEffect(() => {
+    socket.on("message", (data) => {
```

Comment:
worth considering that these `message` events from the server could be regarding ANY of the chatrooms that you're currently in, not just the currently active one